packages/embed/src/factories/create-sidetab/create-sidetab.ts

```diff
@@ -0,0 +1,153 @@
+import { createIframe } from '../../utils'
+
+import { SidetabOptions } from './sidetab-options'
+
+export type Sidetab = {
+  open: () => void
+  close: () => void
+  toggle: () => void
+  refresh: () => void
+  unmount: () => void
+}
+
+const defaultOptions = {
+  buttonColor: '#3a7685',
+  buttonText: 'Launch me',
+}
+
+interface HTMLElementWithParentNode extends HTMLElement {
+  parentNode: Node & ParentNode
+}
+
+const isOpen = (element: HTMLElement): element is HTMLElementWithParentNode => !!element.parentNode
+
+const buildSidetab = () => {
+  const popup = document.createElement('div')
+  popup.className = 'typeform-sidetab'
+  return popup
+}
+
+const buildWrapper = () => {
+  const wrapper = document.createElement('div')
+  wrapper.className = 'typeform-sidetab-wrapper'
+  wrapper.dataset.testid = 'typeform-sidetab-wrapper'
+  return wrapper
+}
+
+const buildSpinner = () => {
+  const spinner = document.createElement('div')
+  spinner.className = 'typeform-spinner'
+  const icon = document.createElement('div')
+  icon.className = 'typeform-sidetab-button-icon'
+  icon.dataset.testid = 'spinner-icon'
+  icon.append(spinner)
+  return icon
+}
+
+const buildTriggerButton = (color: string) => {
+  const button = document.createElement('button')
+  button.className = 'typeform-sidetab-button'
+  button.style.backgroundColor = color
+  return button
+}
+
+const buildTriggerButtonText = (text: string) => {
+  const buttonText = document.createElement('span')
+  buttonText.className = 'typeform-sidetab-button-text'
+  buttonText.innerHTML = text
+  return buttonText
+}
+
+const buildIcon = () => {
+  const icon = document.createElement('div')
+  icon.className = 'typeform-sidetab-button-icon'
+  icon.innerHTML = `<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```

Comment:
Should this support [custom icon like popover](https://github.com/Typeform/embed/pull/195/files#diff-195e7c682997d8f683e65082e6d367b5fac7269a1003a1c2f4926f9416f72d88R58)?